examples/3d/split_screen.rs

```diff
@@ -0,0 +1,103 @@
+//! Renders two cameras to the same window to accomplish "split screen".
+
+use bevy::{
+    core_pipeline::clear_color::ClearColorConfig,
+    prelude::*,
+    render::camera::Viewport,
+    window::{WindowId, WindowResized},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system(set_camera_viewports)
+        .run();
+}
+
+/// set up a simple 3D scene
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // plane
+    commands.spawn_bundle(PbrBundle {
+        mesh: meshes.add(Mesh::from(shape::Plane { size: 100.0 })),
+        material: materials.add(Color::rgb(0.3, 0.5, 0.3).into()),
+        ..default()
+    });
+
+    commands.spawn_scene(asset_server.load("models/animated/Fox.glb#Scene0"));
+
+    // Light
+    commands.spawn_bundle(DirectionalLightBundle {
+        transform: Transform::from_rotation(Quat::from_euler(
+            EulerRot::ZYX,
+            0.0,
+            1.0,
+            -std::f32::consts::FRAC_PI_4,
+        )),
+        directional_light: DirectionalLight {
+            shadows_enabled: true,
+            ..default()
+        },
+        ..default()
+    });
+
+    // camera
+    commands
+        .spawn_bundle(Camera3dBundle {
+            transform: Transform::from_xyz(0.0, 200.0, -100.0).looking_at(Vec3::ZERO, Vec3::Y),
+            ..default()
+        })
+        .insert(LeftCamera);
+
+    // camera
+    commands
+        .spawn_bundle(Camera3dBundle {
+            transform: Transform::from_xyz(100.0, 100., 150.0).looking_at(Vec3::ZERO, Vec3::Y),
+            camera_3d: Camera3d {
+                clear_color: ClearColorConfig::None,
+            },
+            camera: Camera {
+                priority: 1,
+                ..default()
+            },
+            ..default()
+        })
+        .insert(RightCamera);
+}
+
+#[derive(Component)]
+struct LeftCamera;
```

Comment:
In the case of examples, I like having "main" first, immediately followed by the "setup" function, then everything else. Main defines the high level context and setup defines the "scene layout", which seems worth prioritizing so users dont get lost in the weeds. Even in a non-example context this still makes sense to me because if we were to build a "split screen" plugin, we would extract out these components (and the system) and the diff wouldn't involve moving `setup` around.

Reply:
Buuut for really large / complicated setup functions, i see an argument for putting example-related components first, to ensure they get appropriate attention / don't get buried.